### DISCUSSION TITLE
Meaning of the bullet symbol

### DISCUSSION THREAD

**Discussion by patrick-nicodemus** [tree-sitter/tree-sitter#3427] [Q&A] [answered]
### Problem

The documentation here https://tree-sitter.github.io/tree-sitter/creating-parsers is very helpful, however, it is left implicit what the • symbol means. Would it be possible to add a sentence to the documentation which explains what the bullet is?

**Comment by amaanq** [comment]
Sure, I'm pretty sure the bullet is just a separator for the symbols that precede the conflict, and the part after is what the conflict is, as well as what it *could* be reduced to, but doesn't know which one(s) to pick

**Reply by amaanq** [reply]
FYI I converted it to a discussion as I've just pushed a doc update to clarify the meaning of the bullet, and it seems to be a question a user might easily search for in discussions as well, since it's not exactly a bug/usage problem (still is a doc problem technically though)

**Comment by maxbrunsfeld** [comment, marked as answer]
These diagrams represent possible states of the parser. The symbols to the left of the bullet are what the parser has already parsed, and the symbol after the bullet is the "lookahead" symbol. The nature of an LR(1) / GLR(1)  parser is that we must decide how to interpret the preceding symbols given only one token of lookahead.
